platform/admin/core/admin-core-appservice/src/main/java/org/codice/ddf/admin/application/service/impl/ApplicationServiceImpl.java

```diff
@@ -503,27 +503,32 @@ protected Application findFeature(Feature feature, Set<Application> applications
    * @param feature Feature to look for dependencies on.
    * @return A set of all features that are dependencies
    */
-  private Set<Feature> getAllDependencyFeatures(Feature feature) throws Exception {
-    Set<Feature> tmpList = new HashSet<>();
-    // get accurate feature reference from service - workaround for
-    // KARAF-2896 'RepositoryImpl load method incorrectly populates
-    // "features" list'
-    Feature curFeature = featuresService.getFeature(feature.getName(), feature.getVersion());
-
-    if (curFeature != null) {
-      for (Dependency dependencyFeature : curFeature.getDependencies()) {
-        Feature feat =
-            featuresService.getFeature(dependencyFeature.getName(), dependencyFeature.getVersion());
-        if (StringUtils.equals(curFeature.getRepositoryUrl(), feat.getRepositoryUrl())) {
-          tmpList.addAll(getAllDependencyFeatures(feat));
+  private Set<Feature> getAllDependencyFeatures(Feature feature) {
+    try {
+      Set<Feature> tmpList = new HashSet<>();
+      // get accurate feature reference from service - workaround for
+      // KARAF-2896 'RepositoryImpl load method incorrectly populates
+      // "features" list'
+      Feature curFeature = featuresService.getFeature(feature.getName(), feature.getVersion());
+
+      if (curFeature != null) {
+        for (Dependency dependencyFeature : curFeature.getDependencies()) {
+          Feature feat =
+              featuresService.getFeature(
+                  dependencyFeature.getName(), dependencyFeature.getVersion());
+          if (StringUtils.equals(curFeature.getRepositoryUrl(), feat.getRepositoryUrl())) {
+            tmpList.addAll(getAllDependencyFeatures(feat));
+          }
         }
+        tmpList.add(curFeature);
+      } else {
+        // feature may not be installed
+        tmpList.add(feature);
       }
-      tmpList.add(curFeature);
-    } else {
-      // feature may not be installed
-      tmpList.add(feature);
+      return tmpList;
+    } catch (Exception e) {
+      throw new FeatureServiceException("The Karaf Feature Service threw an exception", e);
```

Comment:
Using a RuntimeException to enable the streaming optimization. Impact is negligible because at the root call on the stack we are catching `Exception`.

Reply:
I am on the fence with that one. I do like changing the above to streaming but we are now force to add a try/catch here.

Reply:
This entire module needs work, so I'm okay with it. The finding is fixed.